Integrations/integration-ThreatX.yml

```diff
@@ -0,0 +1,711 @@
+
+    @logger
+    def unwhitelist_ip_command():
+        ip = demisto.args().get('ip', None)
+        results = unwhitelist_ip(ip)
+
+        md = tableToMarkdown('Unwhitelist IP',
+                            results,
+                            ['Result'],
+                            removeNull=True)
+
+        ec = ({
+                'IP.Address(val.Id === obj.Id)': ip
+            })
+
+        demisto.results({
+                'Type': entryTypes['note'],
+                'ContentsFormat': formats['json'],
+                'Contents': results,
+                'HumanReadable': md,
+                'EntryContext': ec
+            })
+
+    @logger
+    def get_entities(entity_name, entity_id, entity_ip):
+        commands = {
+                'command': 'list',
+                'query': {}
+            }
+
+        if entity_name is not None:
+            entity_names = entity_name.split(',')
+            my_entity_name = {'codenames': entity_names}
+            commands['query'].update(my_entity_name)
+
+        if entity_id is not None:
+            entity_ids = entity_id.split(',')
+            my_entity_id = {'entity_ids': entity_ids}
+            commands['query'].update(my_entity_id)
+
+        if entity_ip is not None:
+            entity_ips = entity_ip.split(',')
+            my_entity_ip = {'ip_addresses': entity_ips}
+            commands['query'].update(my_entity_ip)
+
+        return http_request('/entities', commands)
+
+    @logger
+    def get_entity_risk(entity_id):
+        commands = {
+                'command': 'risk_changes',
+                'id': entity_id
+            }
+
+        return http_request('/entities', commands)
+
+    @logger
+    def get_entities_command():
+        entity_name = demisto.args().get('entity_name', None)
+        entity_id = demisto.args().get('entity_id', None)
+        entity_ip = demisto.args().get('entity_ip', None)
+        results = get_entities(entity_name, entity_id, entity_ip)
+        dbot_scores = []
+        output = []
+
+        for entity in results:
+            # Grab the entity risk so we can set the Dbot score for the Actor IPs
+            e_id = entity['id']
+            e_risk = get_entity_risk(e_id)[-1]['risk']
+            entity['risk'] = e_risk
+
+            iplist = []
+            for actor in entity['actors']:
+                # Convert decimal IP to dotted quad
+                ipint = actor['ip_address']
+                ipdot = '.'.join([str(ipint >> (i << 3) & 0xFF) for i in range(4)[::-1]])
+                iplist.append(ipdot)
+                actor['ip_address'] = ipdot
+                # Convert interval_time_start, interval_time_stop, fingerprint.last_seen to pretty time format
+                actor['interval_time_start'] = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(actor['interval_time_start']))
+                actor['interval_time_stop'] = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(actor['interval_time_stop']))
```

Comment:
Please verify `interval_time_stop` is in the dictionary.

Reply:
done